kernel/packages/decentraland-ecs/src/decentraland/Components.ts

```diff
@@ -848,3 +851,72 @@ export class OnAnimationEnd extends OnUUIDEvent<'onAnimationEnd'> {
  */
 @Component('engine.smartItem', CLASS_ID.SMART_ITEM)
 export class SmartItem extends ObservableComponent {}
+
+/**
+ * @public
+ */
+@DisposableComponent('engine.VideoClip', CLASS_ID.VIDEO_CLIP)
+export class VideoClip extends ObservableComponent {
+  @ObservableComponent.readonly
+  readonly url: string
+
+  constructor(url: string) {
+    super()
+    this.url = url
+  }
+}
+
+/**
+ * @public
+ */
+@DisposableComponent('engine.VideoTexture', CLASS_ID.VIDEO_TEXTURE)
+export class VideoTexture extends ObservableComponent {
```

Comment:
Nup... I tried that, apparently you can't extend components

Reply:
Why not? I did this extensively for UI components.

Reply:
> Why not? I did this extensively for UI components.

That's because UIShape is not really a component (It's missing` @DisposableComponent`)
and even if extending `Texture` was possible we need to be able to discriminate them, cause we don't want to enable to add a `VideoTexture` in every component that has a `Texture` field... like for example Material's `bumpTexture` or `refractionTexture`